### PR TITLE
[Core] add ray-core as code-owner for most of the core code-path

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

make https://github.com/orgs/ray-project/teams/ray-core/members become the code-owner on most of core code paths

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scv119** [comment]
@pcmoritz also can you help grant write access of ray-core group? 
image

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -58,15 +58,15 @@
 /java/api/ @jovany-wang @kfstorm @raulchen @ericl @iycheng @WangTaoTheTonic
 
 # C++ worker
-/cpp/include/ray @SongGuyang @raulchen @kfstorm
```
Comment (richardliaw) [inline .github/CODEOWNERS:61]: should this be in addition to ant folks?

**Comment by richardliaw** [comment]
IIRC no… not sure why that is showing up

**Review by rkooo567 — APPROVED** [review]

**Review by fishbone — APPROVED** [review]

**Comment by scv119** [comment]
Let's do it after 2.3 release.

**Pull request merged**